endpoints/counterRoute_test.go

```diff
@@ -156,17 +197,71 @@ func TestCounterHTTPHandlerPOST(t *testing.T) {
 			result.Total, "larger than zero")
 	}
 
+	sqlQuery := `SELECT COUNT(*) AS total FROM counter WHERE page = $1`
+	var counterSummary endpoints.CounterSummary
+	err = db.GetContext(context.Background(), &counterSummary, sqlQuery, page)
+	if err != nil {
+		log.Error().Err(err).Msg("Error getting counter value.")
+		log.Debug().Msgf("SQL query: %s", sqlQuery)
+	}
+
+	if counterSummary.Total != 1 {
+		t.Errorf("handler returned unexpected body: got %v want %d",
+			counterSummary.Total, 1)
+	}
+}
+
+func TestCounterHTTPHandlerPOSTNoPage(t *testing.T) {
+
+	container, err := CreatePostgresTestContainer()
+	if err != nil {
+		t.Errorf("Error creating PostgresTestContainer: %s", err)
+	}
+
+	db, err := StartTestDB(container)
+	if err != nil {
+		t.Errorf("Expected a new database connection, but got %s", err)
+	}
+
+	defer CleanUpTestContainer(container)
+
+	counter := endpoints.NewCounterHandlerContext(db, context.Background(), false)
+
+	rr := httptest.NewRecorder()
+	req, err := http.NewRequest("POST", "v1/counter", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	handler := http.HandlerFunc(counter.CounterHTTPHandler)
+	handler.ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusCreated {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusCreated)
+	}
+
+	var result endpoints.CounterSummary
+	err = json.Unmarshal(rr.Body.Bytes(), &result)
+	if err != nil {
+		t.Errorf("Error unmarshalling response: %s", err)
+	}
+
+	if result.Total != 1 {
+		t.Errorf("handler total returned unexpected body: got %v want %d",
+			result.Total, 1)
+	}
 
 	sqlQuery := `SELECT COUNT(*) AS total FROM counter`
-	var counterSummary counterSummary
+	var counterSummary endpoints.CounterSummary
 	err = db.GetContext(context.Background(), &counterSummary, sqlQuery)
 	if err != nil {
 		log.Error().Err(err).Msg("Error getting counter value.")
 		log.Debug().Msgf("SQL query: %s", sqlQuery)
 	}
 
-	if counterSummary.Total < 1 {
-		t.Errorf("handler returned unexpected body: got %v want %s",
-			counterSummary.Total, "larger than zero")
+	if counterSummary.Total != 1 {
```

Comment:
If I recall correctly, this will likely fail your test case when you run it multiple times.  We only care if the counter is less than 1, indicating the counter is not incrementing and we have an issue. If it is incrementing, no harm in the value being greater than 1 right? 😅

Reply:
Wait, we are using a test container, so we get a new fresh DB instance every time, right? Disregard my comment above 👍🏻